astropy/modeling/tests/test_input.py

```diff
@@ -235,17 +235,17 @@ def test_nonlinear_lsqt_Nset_2d(self, fitter):
             r"Input argument .* does not have the correct dimensions in .* for a model"
             r" set with .*"
         )
+        g2 = models.Gaussian2D(
+            [10, 10],
+            [3, 3],
+            [4, 4],
+            x_stddev=[0.3, 0.3],
+            y_stddev=[0.2, 0.2],
+            theta=[0, 0],
+            n_models=2,
+        )
+        z = g2(self.x.flatten(), self.y.flatten())
```

Comment:
```suggestion
        x = (self.x.flatten(), self.x.flatten())
        y = (self.y.flatten(), self.y.flatten())
        z = g2(x, y)
```
There is a bug in this test which is revealed by these changes. This should fix that bug. The message value will have to change.